src/core/util/grpc_check.h

```diff
@@ -0,0 +1,94 @@
+// Copyright 2025 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_SRC_CORE_UTIL_GRPC_CHECK_H
+#define GRPC_SRC_CORE_UTIL_GRPC_CHECK_H
+
+#ifdef GRPC_POSTMORTEM_CHECKS
+#include <limits.h>
+
+#include "absl/log/log.h"
+#include "src/core/util/postmortem_emit.h"
+
+bool PostMortemEmitAndReturnTrue();
+
+#undef CHECK
```

Comment:
I need to think about this some more... but leaving a comment so I don't forget.

I'm a bit worried about taking the ABSL names, and wonder whether we should instead take GRPC_CHECK etc

What happens if we compile the same header before and after grpc_check.h in different scenarios?
it seems like this might bring ODR violations

Reply:
I see it. Ok, I'll do the refactoring.

Reply:
Macro names have been updated. PTAL.